02-express-tutorial/app.js

```diff
@@ -95,6 +102,102 @@ array.*/
  (which would also be in the public directory.) When you click the button, your JavaScript would 
  issue a fetch call for /api/v1/products. Then you’d add the data you get back to a div in your HTML*/
 
+/*week 4 : The middleware function you create should log the method and url properties from the req object,
+ as well as the current time, before calling next()*/
+ const logger = (req, res, next) => {
+    const currentTime = new Date().toISOString(); // Get current time in ISO format
+    console.log(`[${currentTime}] ${req.method} request to ${req.url}`);
+    next(); // Pass control to the next middleware/route handler
+};
+
+app.use(logger);  // This will log requests for all routes */
+
+// Use the logger middleware for specific routes
+app.get('/', logger, (req, res) => {
+    res.send('Hi There');
+});
+
+week 4:Then, take the logger call out of your app.get() statement, and call it via app.use(), 
+for all paths, instead. Verify that it still works.*/
+// Apply logger middleware to specific paths
+/*app.use(["/about", "/product"], logger); 
+app.get('/about', (req, res) => {
+    res.send('About page');
+});
+
+app.get('/product', (req, res) => {
+    res.send('product page');
+});
+
+//week 4:Now you implement the app.post statement for /api/v1/people
+// Middleware to parse incoming request bodies
+app.use(express.urlencoded({ extended: false }));
+app.use(express.json());  // Parse JSON bodies*/
+
+//for imlementing router and controller
+app.use("/api/v1/people", peopleRouter);
+
+// GET endpoint to return people data
+app.get('/api/v1/people', (req, res) => {
+    res.json(people); // Send the people array as a JSON response
+  });
+
+  // POST endpoint to add a new person
+app.post('/api/v1/people', (req, res) => {
+    const { name } = req.body;
+  
+    // Check if the name was provided in the request body
+    if (!name) {
+      return res.status(400).json({ success: false, message: "Please provide a name" });
+    }
+
+    // Add a new person to the people array
+  const newPerson = { id: people.length + 1, name: req.body.name };
+  people.push(newPerson);
+
+  // Return a success response with the new person's name
+  res.status(201).json({ success: true, name: req.body.name });
+});
+
+
+//Week 4 optional authentication 
+
+// Middleware for parsing JSON bodies and cookies
+app.use(express.json());
```

Comment:
This statement should be put before your route declaration else the body object would not be attached to the req parameter..